apps/core/src/PostHogAnalyticsProvider.tsx

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { PostHogConfig } from 'posthog-js';
+import { PostHogProvider } from 'posthog-js/react';
+import { ReactNode } from 'react';
+
+type PostHogProviderProps = {
+    isEnabled: boolean;
+    children: ReactNode;
+    additionalOptions?: Partial<PostHogConfig>;
+};
+
+export function PostHogAnalyticsProvider({
+    isEnabled,
+    children,
+    additionalOptions,
+}: PostHogProviderProps) {
+    return isEnabled ? (
+        <PostHogProvider
+            apiKey="phc_IggVMJtR5vawlA4H3IIYnIyWjcK8rPiqAI1FlmKZPjp"
```

Comment:
A random q here, is it safe to expose this key or should we put it in an env file.

Reply:
this is a public key so there's no harm in exposing it directly in the source code!